siliconcompiler/tools/vpr/place.py

```diff
@@ -38,7 +41,25 @@ def pre_process(chip):
     step = chip.get('arg', 'step')
     index = chip.get('arg', 'index')
 
-    if not chip.valid('input', 'constraint', 'pins', default_valid=True):
+    part_name = chip.get('fpga', 'partname')
+
+    if 'pinmap' in chip.getkeys('input', 'constraint'):
```

Comment:
`if chip.valid('input', 'constraint', 'pinmap') and chip.get('input', 'constraint', 'pinmap', step=step, index=index):`

Reply:
I tried the proposed change prior to what was implemented and it didn't work; the choice on line 46 matches methodology we greenlit for passing XML pin constraints [here](https://github.com/siliconcompiler/siliconcompiler/blob/2560559abded23d75a5e3f38be395ee993cf9ef9/siliconcompiler/tools/vpr/vpr.py#L155).

I don't have strong opinions about which is "right", but I would prefer to be consistent and to understand what is preventing your proposed change from producing the desired behavior.